Description: Export TMPDIR=/run before running the usb_modeswitch_dispatcher,
 that might be run before /tmp/ is available
Author: Didier Raboud <odyx@debian.org>
Last-Update: 2016-06-13
--- a/usb-modeswitch-upstart.conf
+++ b/usb-modeswitch-upstart.conf
@@ -1,5 +1,6 @@
 start on usb-modeswitch-upstart
 task
+env TMPDIR=/run
 script
 	exec /usr/sbin/usb_modeswitch_dispatcher --switch-upstart $UMS_PARAM
 end script
--- a/usb_modeswitch.sh
+++ b/usb_modeswitch.sh
@@ -47,6 +47,7 @@
 		device_in "link_list" $v_id $p_id
 		if [ "$?" = "1" ]; then
 			if [ -e "/usr/sbin/usb_modeswitch_dispatcher" ]; then
+				export TMPDIR=/run
 				exec usb_modeswitch_dispatcher $1 $2 2>>/dev/null
 			fi
 		fi
--- a/usb_modeswitch@.service
+++ b/usb_modeswitch@.service
@@ -5,4 +5,5 @@
 Type=oneshot
 ExecStart=/usr/sbin/usb_modeswitch_dispatcher --switch-systemd %i
 #ExecStart=/bin/echo %i
+Environment="TMPDIR=/run"
 
